### PR TITLE
HUD Improvements, 1.16 port, fix NanoVG with ARM

### PR DESCRIPTION
**Pull request by Wyvest**
## Description
This pull request brings various HUD improvements to OneConfig, such as positioning improvements and easier handling of HUDs in the HUD editor.
This also adds a work in progress 1.16 port, which should help us advance further for 1.17+ ports.

This also brings minor changes / bug fixes, such as fixing NanoVG with macOS ARM systems and assuring that the position of the mouse is not rounded, fixing some precision issues.

## Related Issue(s)
n/a

## Checklist
- [x] I made a clear description of what was changed
- [x] I stated why these changes were necessary
- [x] I updated documentation or said what needs to be updated
~~- [ ] I made sure these changes are backwards compatible~~
~~- [ ] This pull request is for one feature/bug fix~~

### CODE REVIEW

**Review by nxtdaydelivery — COMMENTED** [review]
will need to test tomorrow, but a look through on my phone detects no breaking changes or anything broken, will approve tommorow

**Comment by DeDiamondPro** [comment]
> will need to test tomorrow, but a look through on my phone detects no breaking changes or anything broken, will approve tommorow

Me whenimage

**Review by DeDiamondPro — DISMISSED** [review]

**Comment by nxtdaydelivery** [comment]
I've had a check over on my phone, will have a proper look tommorow morning now it's done

**Review by DeDiamondPro — APPROVED** [review]

**Review by nxtdaydelivery — COMMENTED** [review]
lgtm

**Review by nxtdaydelivery — DISMISSED** [review]
200 OK

File: versions/src/main/java/cc/polyfrost/oneconfig/platform/impl/LoaderPlatformImpl.java [hunk 1]
```diff
@@ -2,35 +2,52 @@
...
-        //#if MC==11202
-        return new ActiveMod(container.getName(), container.getModId(), container.getVersion());
+            //#if MC==11202
+            return new ActiveMod(container.getName(), container.getModId(), container.getVersion());
+            //#else
+            //$$ return new ActiveMod(container.getModInfo().getDisplayName(), container.getModId(), container.getModInfo().getVersion().getQualifier());
+            //#endif
         //#else
-        //$$ return new ActiveMod(container.getModInfo().getDisplayName(), container.getModId(), container.getModInfo().getVersion().getQualifier());
+        //$$ return null;
```
[the remark this answers is not in the document]
Comment (Wyvest) [reply versions/src/main/java/cc/polyfrost/oneconfig/platform/impl/LoaderPlatformImpl.java:50]: there's no equivalent in fabric

**Review by DeDiamondPro — DISMISSED** [review]
I approve

**Review by DeDiamondPro — DISMISSED** [review]

**Review by DeDiamondPro — APPROVED** [review]

**Review by nxtdaydelivery — COMMENTED** [review]
200 Ok sorry bout delay

**Comment by EthanDevelops** [comment]
JUST MERGE ALREAYD

**Review by nxtdaydelivery — APPROVED** [review]
GitHub moment

**Pull request merged**